### PR TITLE
Azure Github Actions - Create separate GitHub Actions for Unit Test and E2E Deployment 

### PR DESCRIPTION
**Pull request by midhun6989**
Related to the [issue](https://github.com/IBM/zmodstack-deploy/issues/77).

### CODE REVIEW

**Review by ivandov — CHANGES_REQUESTED** [review]
Looks good, but no reason to have duplicated jobs/steps between the two azure github actions. Please remove any of the steps that you've copied over into the new action from the old action, and also remove the `branches-ignore` from the new action.

**Comment by midhun6989** [comment]
> Looks good, but no reason to have duplicated jobs/steps between the two azure github actions. Please remove any of the steps that you've copied over into the new action from the old action, and also remove the `branches-ignore` from the new action.

@ivandov Updated as mentioned.

**Review by ivandov — APPROVED** [review]
LGTM

File: .github/workflows/azure-test.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+name: Azure Test
+
+on:
+  push:
+    branches-ignore:
+      - main
+      - dev
+      - release-*.*.*
```
[the remark this answers is not in the document]
Comment (midhun6989) [reply .github/workflows/azure-test.yml]: Updated.

File: .github/workflows/azure-test.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+name: Azure Test
```
[the remark this answers is not in the document]
Comment (midhun6989) [reply .github/workflows/azure-test.yml]: Updated.

File: .github/workflows/azure-test.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        bicepFilePath: ${{ env.ARM_TEMPLATE_BICEP }}
+        outputFilePath: ${{ env.ARM_TEMPLATE_BICEP_TO_JSON }} 
+    - name: Check mainTemplate.json is Latest
+      run: diff ${{ env.ARM_TEMPLATE_JSON }} ${{ env.ARM_TEMPLATE_BICEP_TO_JSON }}
+      shell: bash
+      continue-on-error: true  
+    - name: Run ARM-TTK
+      uses: microsoft/action-armttk@v1
+      with:
+        workdir: ${{ env.WORK_DIR }}
```
[the remark this answers is not in the document]
Comment (midhun6989) [reply .github/workflows/azure-test.yml:33]: Updated.

**Pull request merged**